diff --git a/src/contour/CMakeLists.txt b/src/contour/CMakeLists.txt
index 86c8e61f..c6825f07 100644
--- a/src/contour/CMakeLists.txt
+++ b/src/contour/CMakeLists.txt
@@ -206,11 +206,11 @@ if(FREEBSD)
     target_link_directories(contour PUBLIC "/usr/local/lib")
 endif()
 
-if (TARGET yaml-cpp::yaml-cpp)
-    set(YAML_CPP_LIBRARIES yaml-cpp::yaml-cpp)
-else()
-    set(YAML_CPP_LIBRARIES yaml-cpp)
-endif()
+#if(YAML_CPP_INCLUDE_DIR)
+    set(YAML_CPP_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../_deps/sources/yaml-cpp/include")
+    include_directories(${YAML_CPP_INCLUDE_DIR})
+    target_include_directories(contour PRIVATE ${YAML_CPP_INCLUDE_DIR})
+#endif()
 
 target_link_libraries(contour
     PRIVATE
@@ -218,7 +218,7 @@ target_link_libraries(contour
     crispy::core
     vtbackend
     vtrasterizer
-    ${YAML_CPP_LIBRARIES}
+    yaml-cpp
 )
 
 # {{{ GUI: Declare Qt build dependencies
